volumegroup/README.md

```diff
@@ -0,0 +1,455 @@
+
+This RPC will be called by the CO to create a new volume group on behalf of a user. This operation MUST be idempotent.
+If a volume group corresponding to the specified volume group name already exists, is compatible with the specified
+parameters in the CreateVolumeGroupRequest, the Plugin MUST reply 0 OK with the corresponding CreateVolumeGroupResponse.
+CSI Plugins MAY create the following types of volume groups:
+
+Create a new empty volume group. Note that N volumes with some backend label Y could be considered to be in "group Y"
+which might not be a physical group on the storage backend. In this case, an empty group can still be created by the CO
+to hold volumes. After the empty group is created, create a new volume, specifying the group name in the volume.
+
+At restore time, create a single volume from individual snapshot and then join an existing group. Create an empty group.
+Create a volume from snapshot, specifying the group name in the volume.
+
+Future goals:
+Create a new volume group from a source group snapshot. Create a new volume group from a source group. Create a new
+volume group and add a list of existing volumes to the group.
+
+Non goal:
+Create a new group and at the same time create a list of new volumes in the group.
+
+```protobuf
+message CreateVolumeGroupRequest {
+  option (vg_alpha_message) = true;
+
+  // suggested name for volume group (required for idempotency)
+  // This field is REQUIRED.
+  string name = 1;
+
+  // params passed from VolumeGroupClass
+  // This field is OPTIONAL.
+  map<string, string> parameters = 2;
+
+  // Secrets required by plugin to complete volume group creation
+  // request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  map<string, string> secrets = 3 [(vg_csi_secret) = true];
+}
+
+message CreateVolumeGroupResponse {
+  option (vg_alpha_message) = true;
+
+  // Contains all attributes of the newly created volume group.
+  // This field is REQUIRED.
+  VolumeGroup volume_group = 1;
+}
+
+message VolumeGroup {
+  option (vg_alpha_message) = true;
+
+  // The identifier for this volume group, generated by the plugin.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Opaque static properties of the volume group.
+  // This field is OPTIONAL.
+  map<string, string> volume_group_context = 2;
+
+  // Underlying volumes in this group. The same definition in CSI
+  // Volume.
+  // This field is OPTIONAL.
+  // To support the creation of an empty group, this list can be empty.
+  // However, this field is not empty in the following cases:
+  // - Response from ListVolumeGroups or ControllerGetVolumeGroup if the
+  //   VolumeGroup is not empty.
+  // - Response from ModifyVolumeGroupMembership if the
+  //   VolumeGroup is not empty after modification.
+  repeated VgVolume volumes = 3;
+}
+
+// Information about a specific volume.
+message VgVolume {
```

Comment:
Can you re-use the CSI Volume message here?

ReclaimSpace uses `csi.v1.VolumeCapability` from the CSI spec, hopefully you can import the CSI types ins a similar way. See https://github.com/csi-addons/spec/blob/main/reclaimspace/README.md

You can mark secrets with `csi.v1.csi_secret` too then (see the fencing README.md for an example).

Reply:
You are now importing `csi.proto`, so you can use `csi.v1.Volume` instead of `VgVolume`.

Reply:
done